airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/message/DestinationMessageDeserializer.kt

```diff
@@ -25,7 +25,28 @@ class DefaultDestinationMessageDeserializer(private val messageFactory: Destinat
             val airbyteMessage = serialized.deserializeToClass(AirbyteMessage::class.java)
             return messageFactory.fromAirbyteMessage(airbyteMessage, serialized)
         } catch (t: Throwable) {
-            throw RuntimeException("Failed to deserialize AirbyteMessage")
+            /**
+             * We don't want to expose client data, but we'd like to get as much info as we can
+             * about these malformed messages.
+             */
+            val type =
+                if (serialized.contains("RECORD")) {
```

Comment:
You might want something like `"type":"RECORD"` to help avoid false positives, but ymmv